File: src/main/java/com/github/kalcoder/plugi/commands/CommandPermission.java
package com.github.kalcoder.plugi.commands;

import org.bukkit.permissions.Permission;

public enum CommandPermission {

//  Admin
  ADMIN_ALL,
  ADMIN_PLUGICONFIG,
  ADMIN_SUDO,
  
//  Players
  PLAYERS_ALL,
  PLAYERS_INVSEE;
  
  private Permission permission;
  
  CommandPermission() {
    String permissionName = name().toLowerCase().replaceAll("_", ".").replaceAll("all", "*");
    System.out.println(permissionName);
    permission = new Permission(permissionName);
  }
  
  public Permission getPermission() {
    return permission;
  }
}

File: src/main/kotlin/com/github/kalcoder/plugi/commands/CommandPermission.kt
package com.github.kalcoder.plugi.commands

import org.bukkit.permissions.Permission

enum class CommandPermission {

  //  Admin
  ADMIN_ALL,
  ADMIN_PLUGICONFIG,
  ADMIN_SUDO,

  //  Players
  PLAYERS_ALL,
  PLAYERS_INVSEE;

  val permission: Permission

  init {
    val permissionName = name.toLowerCase().replace("_".toRegex(), ".").replace("all".toRegex(), "*")
    println(permissionName)
    permission = Permission(permissionName)
  }

}